app/OpenGraphImage.php

```diff
@@ -25,32 +44,38 @@ public static function makeFileName($uniqueId, $name): string
         return "{$uniqueId}_" . Str::slug($name, '-') . '.png';
     }
 
-    public function __construct($fileName)
+    public function generate()
     {
-        $this->fileName = $fileName;
+
+        if (! $this->makeStorageDirectory()) {
+            $this->deleteImagesForProject();
+        }
+
+        $image = $this->make();
+        $this->save($image);
     }
 
-    public function storageDirectory(): string
+    protected function storageDirectory(): string
     {
         return config('opengraph.image_directory_name') . '/';
     }
 
-    public function storagePath(): string
+    protected function storagePath(): string
     {
         return Storage::disk('public')->path('') . $this->storageDirectory() . "{$this->fileName}";
     }
 
-    public function makeStorageDirectory(): bool
+    protected function makeStorageDirectory(): bool
     {
         if (Storage::exists($this->storageDirectory())) {
             return false;
         }
 
-        Storage::makeDirectory($this->storageDirectory());
+        Storage::ensureDirectoryExists($this->storageDirectory());
```

Comment:
This call would actually replace the whole `makeStorageDirectory()` function; it checks if it exists, and creates it if it doesn't.